client/js/src/main/scala/org/http4s/client/NodeJSClientBuilder.scala

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2014 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package client
+
+import cats.effect.kernel.Async
+import cats.effect.kernel.Resource
+import cats.effect.std.Dispatcher
+import cats.syntax.all._
+import org.http4s.internal.BackendBuilder
+import org.http4s.nodejs.ClientRequest
+import org.http4s.nodejs.IncomingMessage
+
+import scala.annotation.nowarn
+import scala.scalajs.js
+import scala.scalajs.js.JSConverters._
+
+import NodeJSClientBuilder._
+
+/** Builder for a [[Client]] backed by Node.js [[https://nodejs.org/api/http.html#httprequestoptions-callback http.request]].
+  *
+  * The Node.js client adds no dependencies beyond `http4s-client`.
+  * This client is currently not production grade, but convenient for
+  * experimentation and exploration.
```

Comment:
Also FTR, I wouldn't be opposed to keeping this http4s private or just dumping it altogether. It's useful for testing, but dealing with JS APIs is fiddly and I don't want it to become a maintenance burden on the core team especially when the whole point of the schism is to reduce said burden.

I don't expect it to break (the underlying Node.js APIs are stable) but it might require additional work if/when the `ClientRouteTestBattery` gets more tests.

Reply:
Is it useful in a REPL?  That's the other place that the Java 8 one comes in handy.

Reply:
No, because there is no Scala.js REPL. This is on scala-cli's radar but seems unlikely.
- https://github.com/VirtusLab/scala-cli/issues/783